BFS/ccconac/2583.js

```diff
@@ -0,0 +1,65 @@
+const input = require('fs')
+  .readFileSync('text.txt')
+  .toString()
+  .trim()
+  .split('\n');
+const [M, N, K] = input.shift().split(' ').map(Number);
+
+const graph = Array.from(Array(M), () => Array(N).fill(0));
+const answer = [];
+
+// 입력값을 그래프로 변환하기 위한 반복문
+for (let i = 0; i < K; i += 1) {
+  const [x1, y1, x2, y2] = input[i].split(' ').map(Number);
+
+  // 직사각형 영역은 1로 표시, 빈 영역은 0으로 표시
+  for (let y = M - y2; y < M - y1; y += 1) {
+    for (let x = x1; x < x2; x += 1) {
+      graph[y][x] = 1;
+    }
+  }
+}
+
+function bfs(start) {
+  const ds = [
+    [-1, 0],
+    [1, 0],
+    [0, 1],
+    [0, -1],
+  ]; // 좌우상하
+  const queue = [start]; // 시작점은 무조건 돌아야 하는 노드이기 때문에 큐에 넣고 시작
+
+  let count = 0;
+
+  while (queue.length) {
+    // 큐에 남아 있는 노드가 없다면 탐색이 종료된 것
+    const [cy, cx] = queue.shift();
+    count += 1;
+
+    for (let i = 0; i < 4; i += 1) {
+      const ny = cy + ds[i][1];
+      const nx = cx + ds[i][0];
```

Comment:
제가 요즘 자주 사용하는 방법인데요! 
```suggestion
for(let [dx, dy] of ds) {
   const nx = cx + dx;
   const ny = cy + dy;
}
```
로 적는 건 어떤가요?! `dx` 배열과 `dy` 배열을 합쳐서 `ds`라는 2차원을 만드셨길래 말씀드려요!!

Reply:
```suggestion
const [dx,dy] = [
  [1,-1,0,0],
  [0,0,1,-1]
];

for (let i = 0; i < 4; i += 1) {
  const nx = cx + dx[i];
  const ny = cy + dy[i];
}
```

저는 dx,dy를 정의할때 구조분해할당을 이용하는 편입니다!